src/frontends/onnx/frontend/src/detail/subgraph_extraction.cpp

```diff
@@ -283,17 +283,42 @@ SubgraphExtractor::SubgraphExtractor(ONNX_NAMESPACE::GraphProto& graph)
     }
 }
 
-void SubgraphExtractor::add_new_inputs(const std::vector<InputEdge>& new_inputs) {
-    for (const auto& input_edge : new_inputs) {
+void SubgraphExtractor::add_new_inputs(const std::vector<InputEdge>& new_inputs, const bool merge_inputs) {
+    if (merge_inputs) {
+        const auto& input_edge = new_inputs[0];
         const auto tensor_name = get_input_tensor_name(m_onnx_graph, input_edge);
 
-        // in case an edge is connected to a single node, a new graph input should be added
-        // and connected to that node; the new edge is an edge between the node and new input
-        const auto new_input = append_new_graph_input(m_onnx_graph, input_edge, m_tensor_consumers[tensor_name]);
+        int input_consumers = m_tensor_consumers[tensor_name];
+        if (m_tensor_consumers[tensor_name] - new_inputs.size() == 0) {
+            input_consumers = 0;
+        }
```

Comment:
Let's try
`int input_consumers = m_tensor_consumers[tensor_name] - new_inputs.size();`

Reply:
algoritm is completetly changed, no need to use proposed equation.